LibertyDSNP/example-client
# add linter and tests to github actions

# Problem

As a developer, I want to be able to be sure that my tests and lints all pass via CI 
[link to Pivotal Tracker #177350710](https://www.pivotaltracker.com/story/show/177350710)

# Solution

- created a `.github/workflows/main.yml` file that stores the config for the GitHub actions tests. 

## Change summary:

- created the `main.yml` file and populated it with the same tests and linters that we are running for the sdk.
- debugged the error messages
 - adjusted the package config of jest, enzyme, and jest-junit to get things working.
 - updated the counter and counter.test to work with new testing 
 - added `"quotes": ["error", "double"],` to the linter config so it didn't throw an error on build
 - added the jest config directory

## Steps to Verify:

1. Check that GitHub Actions tests are failing when they should
2. Check that they are passing when they should

NOTE:
Enzyme is not ready for react 17 yet, so using an unofficial version and installing packages with `npm i --legacy-peer-deps` if using node v7.
- Change to official version of enzyme when correct version is available
- Issue: https://github.com/enzymejs/enzyme/issues/2429
